lib/charms/tls_certificates_interface/v0/tls_certificates.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the tls-certificates relation.
+
+This library contains the Requires and Provides classes for handling
+the tls-certificates interface.
+
+## Getting Started
+
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.tls_certificates_interface.v0.tls_certificates
+```
+
+You will also need to add the following library to the charm's `requirements.txt` file:
+- jsonschema
+
+### Provider charm
+Example:
+```python
+from charms.tls_certificates_interface.v0.tls_certificates import (
+    Cert,
+    InsecureCertificatesProvides,
+)
+from ops.charm import CharmBase
+
+
+class ExampleProviderCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.insecure_certificates = InsecureCertificatesProvides(self, "certificates")
```

Comment:
`InsecureCertificateProvides` isn't a great name ;-) is the naming due to the fact they're self-signed?

Reply:
I replaced instances of "Insecure" with tls.